### PR TITLE
Fix BlockForRpc serialization for different engines

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs [hunk 1]
```diff
@@ -37,19 +38,16 @@ public LongConverter(NumberConversion conversion)
...
-                    writer.WriteValue(value.ToHexString(true));
+                    writer.WriteValue(value == 0L ? "0x0" : value.ToHexString(true));
                     break;
                 case NumberConversion.Decimal:
-                    writer.WriteValue(value.ToString());
+                    writer.WriteValue(value == 0 ? "0" : value.ToString());
+                    break;
+                case NumberConversion.Raw:
+                    writer.WriteValue(value);
                     break;
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs]: can you test it in an array? we had issues with serializing arrays when using WriteValue from Json.NET

**Comment by tkstanczak** [comment]
where are trace module tests? won't it fail?

File: src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs]: what scenario exactly this is just a primitive?

**Comment by LukaszRozmej** [comment]
There wern't any :/ but serialization is fine

File: src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs (hunk 1 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs]: you can leave it -unlikely to happen, Parity trace has an array of ints but it is handled separately

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Serialization.Json/LongConverter.cs]: unlikely we would end up with an array if nullable longs

**Pull request merged**